testing/file_test/file_test_base.cpp

```diff
@@ -517,62 +517,80 @@ struct SplitState {
   int file_index = 0;
 };
 
-// Reformats `[[@LSP:` and `[[LSP-NOTIFY:` as an LSP call with headers. For
-// notifications, `lsp_call_id` is null.
+// Reformats `[[@LSP:` and similar keyword as an LSP call with headers.
 static auto ReplaceLspKeywordAt(std::string* content, size_t keyword_pos,
-                                int* lsp_call_id, llvm::StringLiteral keyword,
-                                llvm::StringLiteral method_or_id_label)
-    -> ErrorOr<size_t> {
-  auto method_or_id_start = keyword_pos + keyword.size();
+                                int& lsp_call_id) -> ErrorOr<size_t> {
+  llvm::StringRef content_at_keyword =
+      llvm::StringRef(*content).substr(keyword_pos);
+
+  auto [keyword, body_start] = content_at_keyword.split(":");
+  if (body_start.empty()) {
+    return ErrorBuilder() << "Missing `:` for `"
+                          << content_at_keyword.take_front(10) << "`";
+  }
+
+  // Whether the first param is a method or id.
+  llvm::StringRef method_or_id_label = "method";
+  // Whether to attach the `lsp_call_id`.
+  bool use_call_id = false;
+  // The JSON label for extra content.
+  llvm::StringRef extra_content_label;
+  if (keyword == "[[@LSP-CALL") {
+    use_call_id = true;
+    extra_content_label = "params";
+  } else if (keyword == "[[@LSP-NOTIFY") {
+    extra_content_label = "params";
+  } else if (keyword == "[[@LSP-REPLY") {
+    method_or_id_label = "id";
+    extra_content_label = "result";
+  } else if (keyword != "[[@LSP") {
+    return ErrorBuilder() << "Unrecognized @LSP keyword at `"
+                          << content_at_keyword.take_front(10) << "`";
```

Comment:
Seems like this should stop at the colon
```suggestion
    return ErrorBuilder() << "Unrecognized @LSP keyword `"
                          << keyword.drop_front(2) << "`";
```
or if you are really worried about it being too long, use a longer limit since "@LSP-NOTIFY" is more than 10 chars.
```suggestion
    return ErrorBuilder() << "Unrecognized @LSP keyword `"
                          << keyword.drop_front(2).take_front(16) << "`";
```

Reply:
I'll change this to the keyword value front. Note line 615 does not drop the `[[` from the keyword; I plan to keep similar behavior here.